android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt

```diff
@@ -22,6 +22,19 @@ internal const val URL_TYPE_SUGGEST_JSON = "application/x-suggestions+json"
 internal const val URL_TYPE_SEARCH_HTML = "text/html"
 internal const val URL_REL_MOBILE = "mobile"
 internal const val IMAGE_URI_PREFIX = "data:image/png;base64,"
+internal const val GOOGLE_ID = "google"
+
+// List of general search engine ids, taken from
+// https://searchfox.org/mozilla-central/rev/ef0aa879e94534ffd067a3748d034540a9fc10b0/toolkit/components/search/SearchUtils.sys.mjs#200
+internal val GENERAL_SEARCH_ENGINE_IDS = setOf(
```

Comment:
It seems there are other "general" search engines in `list.json`.
At a quick look there's at least Cốc Cốc[ for vietnamese](https://github.com/mozilla-mobile/firefox-android/blob/7fe93f305b499e01e749748fdc27981bff770187/android-components/components/feature/search/src/main/assets/search/list.json#L880) - https://coccoc.com/search.
I think we should "manually" go through the current search engines from `list.json` and ensure all general ones are added to this list.